src/main/java/com/petros/bibernate/dao/EntityPersister.java

```diff
@@ -213,13 +217,29 @@ private static String getUpdatePlaceholders(List<?> updatableColumns) {
                 .collect(Collectors.joining(", "));
     }
 
-    private static <T> T mapResultSetToEntity(Class<T> entityClass, ResultSet resultSet) {
+    private <T> T mapResultSetToEntity(Class<T> entityClass, ResultSet resultSet) {
         try {
             T entity = entityClass.getConstructor().newInstance();
             for (var entityField : entityClass.getDeclaredFields()) {
                 entityField.setAccessible(TRUE);
-                String columnName = getColumnName(entityField);
-                entityField.set(entity, resultSet.getObject(columnName));
+
+                if (isRegularField(entityField)) {
+                    if (!hasEntityRelation(entityField)) {
```

Comment:
Why do we skip any field marked with @Id or @Column?

Reply:
Mistake. Fixed in a new commit